pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,385 @@
+    printf "\t%-25s %s\n" "--mappings" "Check mappings/fields count & limit";
+    printf "\t%-25s %s\n" "--user-roles" "Check configured user roles";
+    printf "\t%-25s %s\n" "--ism" "Check ISM";
+    printf "\t%-25s %s\n" "--object-store-access" "Check object store access";
+    printf "\t%-25s %s\n" "--fluentd" "Check that fluentd can connect to opensearch";
+    printf "%s\n" "[NOTE] If no argument is specified, it will go over all of them.";
+
+    exit 0;
+}
+
+function sc_opensearch_checks(){
+    if [[ ${#} == 0 ]]; then
+        check_opensearch_cluster_health
+        check_opensearch_snapshots_status
+        check_opensearch_breakers
+        check_opensearch_indices
+        check_opensearch_aliases
+        check_opensearch_mappings
+        check_opensearch_user_roles
+        check_opensearch_ism
+        check_object_store_access
+        check_fluentd_connection
+        exit 0
+    fi
+    while [[ ${#} -gt  0 ]]; do
+      case ${1} in
+          --cluster-health )
+              check_opensearch_cluster_health
+          ;;
+          --snapshot-status )
+              check_opensearch_snapshots_status
+          ;;
+          --breakers )
+              check_opensearch_breakers
+          ;;
+          --indices )
+              check_opensearch_indices
+          ;;
+          --aliases )
+              check_opensearch_aliases
+          ;;
+          --mappings )
+              check_opensearch_mappings
+          ;;
+          --user-roles )
+              check_opensearch_user_roles
+          ;;
+          --ism )
+              check_opensearch_ism
+          ;;
+          --object-store-access )
+              check_object_store_access;
+          ;;
+          --fluentd )
+              check_fluentd_connection;
+          ;;
+          --help )
+            opensearch_check_help
+          ;;
+      esac
+      shift
+    done
+}
+
+check_opensearch_cluster_health() {
+  echo -ne "Checking if opensearch cluster is healthy ... "
+  cluster_health=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cluster/health")
+  status=$(echo "$cluster_health" | jq -r '.status')
+  if [[ $status != "green" ]]; then
+    echo -e "failure ❌";
+    echo "$cluster_health" | jq
+  else
+    echo -e "success ✔";
+  fi
+}
+
+check_opensearch_snapshots_status(){
+  echo -ne "Checking opensearch snapshots status ... "
+  repo_name=$(yq4 -e '.opensearch.snapshot.repository' "${config['config_file_sc']}")
+  repo_exists_status=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_snapshot/${repo_name}" | jq "select(.error)")
+  if [[ -z "$repo_exists_status" ]]; then
+    if kubectl get "cronjob" -n "opensearch-system" "opensearch-backup" &> /dev/null
+    then
+        last_schedule=$(kubectl get cronjob -n opensearch-system opensearch-backup -ojsonpath='{.status.lastScheduleTime}')
+        if [[ $last_schedule != $(date +"%Y-%m-%dT12:00:00Z") ]]; then
```

Comment:
This check will not be reliable as the time may be different. Could you also consider to add a check of the current snapshots via `_cat/snapshots/$repo_name` so that the snapshots have completed and that the last is within the configured snapshot interval?

Reply:
Sure